iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/hsm/IotHubSasTokenHsmAuthenticationProvider.java

```diff
@@ -169,7 +179,23 @@ private static IotHubSasToken createNewSasToken(
             String data = audience + "\n" + expiresOn;
             String signature = signatureProvider.sign(moduleId, data, generationId);
 
-            String host = gatewayHostName != null && !gatewayHostName.isEmpty() ? gatewayHostName : hostname;
+            if (gatewayHostName != null && !gatewayHostName.isEmpty()
```

Comment:
This logic looks like it could sit at a higher level since this class inherits from another. Could you look into moving this up a bit? That way all these AuthenticationProvider classes don't need this same logic within them

Reply:
I do have the same [logic](https://github.com/Azure/azure-iot-sdk-java/blob/brycewang/MqttGatewayHostName/iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/IotHubAuthenticationProvider.java#L35-L39) in the ctors of base class inherited by other AuthenticationProvider classes. This logic right here is in creating new SAS token, which is being used only by this class. I prefer to leave it as-is.